src/lib/components/Swap/Input.tsx

```diff
@@ -81,11 +81,24 @@ export default function Input({ disabled, focused }: InputProps) {
     return insufficientBalance ? 'error' : undefined
   }, [balance, inputCurrencyAmount, swapInputCurrencyAmount])
 
+  const amount = useMemo(() => {
```

Comment:
This is the same between `Input` and `Output`. Can you make it an exported hook from `Input` and reuse it in `Output`? See `LoadingRow`, `Balance` as prior work.